data_source_ssh_tunnel.go

```diff
@@ -120,49 +182,22 @@ func dataSourceSSHTunnelRead(d *schema.ResourceData, meta interface{}) error {
 		effectiveAddress := localListener.Addr().String()
 		if effectiveAddress != localAddress {
 			log.Printf("[DEBUG] localAddress: %v", effectiveAddress)
-			d.Set("local_address", effectiveAddress)
+			err = d.Set("local_address", effectiveAddress)
+			if err != nil {
+				panic(err)
+			}
 		}
 
 		lastColon := strings.LastIndex(effectiveAddress, ":")
 		port := effectiveAddress[lastColon+1 : len(effectiveAddress)]
 		log.Printf("[DEBUG] port: %v", port)
-		d.Set("port", port)
+		err = d.Set("port", port)
+		if err != nil {
+			panic(err)
+		}
 
-		go func() {
-			sshClientConn, err := ssh.Dial("tcp", host, sshConf)
-			if err != nil {
-				panic(err)
-			}
+		go forward(d, host, remoteAddress, localListener, sshConf)
```

Comment:
I was thinking about adding some tests to it, so moved it to named function. And I do like the convention :)

Reply:
Well, if it's all the same to you, can you change it back so that this PR only adds the support for the SSH agent?

We can have a philosophical discussion about code conventions and tests in a separate PR. Personally, I am of the opinion that this project is not big enough, or complicated enough, to merit tests at this point.

Reply:
I'll try to rewrite it back, but I think we should at least keep the copyConn (line99) to not duplicate code.

Reply:
Rewritten back, to the original shape :)

Reply:
Hey @stefansundin 
So should I open a new pull request to accept the changes made so far? 

Thanks.

Reply:
Hi. What purpose would that serve?

I haven't had time to review this again, but I am hoping I will get the time to do so this weekend.

Reply:
A new pull request? I think none actually. 
Honestly I just wanted to check if I’ve still got your attention :)

Reply:
Hey @stefansundin

Hope you aren doing great!
Have you head a chance to look at the changes?

Thanks